File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
  const hashobj = {};
  // O(n) 복잡도로 해결 -> 해시테이블 이용, 키:이름, 값:동명이인 수
  for (const p of participant) {
    if (hashobj[p]) {
      hashobj[p]++;
    } else {
      hashobj[p] = 1;
    }
  }
  for (const c of completion) {
    hashobj[c]--;
  }
  for (const name of participant) {
    if (hashobj[name] > 0) {
      return name;
    }
  }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.io.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Map<String, Integer> m = new HashMap<>();
        for(String name : participant){
            m.put(name, m.getOrDefault(name, 0) + 1);
        }
        
        for(String name : completion){
            if(m.containsKey(name)){
                m.put(name, m.get(name)-1);
            }
        }
        
        for(String name: m.keySet()){
            if(m.get(name) != 0){
                answer = name;
                break;
            }
        }
        
        return answer;
    }
}